planner/optimize.go

```diff
@@ -220,6 +220,46 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 	return bestPlan, names, nil
 }
 
+func allowInReadOnlyMode(sctx sessionctx.Context, node ast.Node) (bool, error) {
+	pm := privilege.GetPrivilegeManager(sctx)
+	if pm == nil {
+		return true, nil
+	}
+	roles := sctx.GetSessionVars().ActiveRoles
+	// allow replication thread
+	// NOTE: it is required, whether SEM is enabled or not, only user with explicit RESTRICTED_REPLICA_WRITER_ADMIN granted can ignore the restriction, so we need to surpass the case that if SEM is not enabled, SUPER will has all privileges
+	if pm.HasExplicitlyGrantedDynamicPriviledge(roles, "RESTRICTED_REPLICA_WRITER_ADMIN", false) {
```

Comment:
The name `RESTRICTED_` is used to say that it is not implied by root when SEM is enabled. Should we rename it to just `REPLICA_WRITER_ADMIN` since it now has no relation to SEM and does an explicit privilege check?

Reply:
The behavior of it is exactly the same as `RESTRICTED_` variables when SEM is enabled (i.e. "not implied by root when SEM is enabled"), adding extra check is to make sure it is no implied by root when SEM is not enabled, which I think is more restrictive than `RESTRICTED_`, so I would like:

1. either call it `RESTRICTED_`
2. call it `EXPLICIT_`

But not just `REPLICA_WRITER_ADMIN`.

What do you think?